draft-ietf-ipsecme-ikev2-multiple-ke-00.xml

```diff
@@ -885,6 +885,129 @@ Type     Description                   Used In
     	&RFC8391;
     </references>
 
+  <section title="Sample Multiple Key Exchanges" anchor="sample-exchanges">
+    <t>This appendix shows some examples of multiple key exchanges.  These examples are purely for information
+    purposes and they describe some message flow scenarios that may occur in establishing an IKE or CHILD SA.  Note
+    that some payloads that are not relevant to multiple key exchanges may be omitted for brevity.
+    </t>
+
+    <section title="No Additional Key Exchange Used" anchor="sample-exchanges-none-selected">
+      <t>The initiator proposes two sets of optional additional key exchanges, but the responder does not
+      support any of them.  The responder chooses NONE for each set and consequently, IKE_INTERMEDIATE
+      exchange does not takes place and the exchange proceeds to IKE_AUTH phase.  The resulting keying
+      materials are the same as those derived with <xref target="RFC7296"/>.</t>
+
+      <figure><artwork align="center"><![CDATA[
+Initiator                     Responder
+------------------------------------------------------------------------
+HDR(IKE_SA_INIT), SAi1(.. AKE*...), --->
+KEi1, Ni, N(IKEV2_FRAG_SUPPORTED),
+N(INTERMEDIATE_EXCHANGE_SUPPORTED)
+  Proposal #1
+    Transform ECR (ID = ENCR_AES_GCM_16,
+                   256-bit key)
+    Transform PRF (ID = PRF_HMAC_SHA2_512)
+    Transform KE (ID = Curve25519)
+    Transform AKE1 (ID = PQ_KEM_1)
+    Transform AKE1 (ID = PQ_KEM_2)
+    Transform AKE1 (ID = NONE)
+    Transform AKE2 (ID = PQ_KEM_3)
+    Transform AKE2 (ID = PQ_KEM_4)
+    Transform AKE2 (ID = NONE)
+                         <--- HDR(IKE_SA_INIT), SAr1(.. AKE*...),
+                              KEr1, Nr, N(IKEV2_FRAG_SUPPORTED),
+                              N(INTERMEDIATE_EXCHANGE_SUPPORTED)
+                                Proposal #1
+                                  Transform ECR (ID = ENCR_AES_GCM_16,
+                                                 256-bit key)
+                                  Transform PRF (ID = PRF_HMAC_SHA2_512)
+                                  Transform KE (ID = Curve25519)
+                                  Transform AKE1 (ID = NONE)
+                                  Transform AKE2 (ID = NONE)
+
+HDR(IKE_AUTH), SK{ IDi, AUTH, SAi2, TSi, TSr } --->
+                         <--- HDR(IKE_AUTH), SK{ IDr, AUTH, SAr2,
+                              TSi, TSr }
+      ]]></artwork></figure>
+    </section>
+
+    <section title="Additional Key Exchange for CHILD SA only" anchor="sample-exchanges-ake-child-sas">
+      <t>The exchanges below show that the initiator does not propose the use of additional key exchanges
+      to establish an IKE SA, but they are required in order to establish a CHILD SA.  Note that the
```

Comment:
s/CHILD SA/Child SA
Note, that the very first Child SA will be created as result of IKE_AUTH, so if both the initiator and the responder have policy to create only quantum-protected IPsec SA, then both should exchange CHILDLESS_IKEV2_SUPPORTED in the IKE_SA_INIT and don't create ESP SA in IKE_AUTH.

A few words should be added that IKE SA is then immediately rekeyed and then Child SAs are creared as usual.

Reply:
> HI CJ, thank you for this diagrams. I wrote some comments. I may directly edit the file if you agree.

Hi @smyslov, many thanks for your review. Please go ahead and edit the file directly :-)